src/Pages/License/Generate/GenerateLicense.jsx

```diff
@@ -32,30 +35,57 @@ const renderSwitch = (count) => {
     return <FeatureDetails />;
   case 3:
     return <LimitationDetails />;
+  case 4:
+    return <OptionalDetails />;
   }
 };
 
+const noOfKeys = (keyAmount) => {
+  if(keyAmount < 1){
+    return 1;
+  }
+  return keyAmount;
+}
+
 const GenerateLicense = (props) => {
   const [count, setCount] = useState(0);
   const detailsPage = renderSwitch(count);
-  
+  const navigate = useNavigate();
+  const { state } = useContext(AppContext);
+  const btnDisabled = state.error;
+
   return (
     <>
       <Grid item container direction="column" {...props}>
         {detailsPage}
         <Stack direction="row" justifyContent="space-between">
-          {count > 0 ? (
-            <Button variant="text" onClick={() => setCount(count - 1)}>
+          {count > 0 && count < 5 ? (
+            <Button variant="text" disabled={btnDisabled} onClick={() => {setCount(count - 1)}}>
               Go Back
             </Button>
           ) : (
             <div style={{ display: "block" }} />
           )}
-          {count < 3 && (
-            <Button variant="contained" onClick={() => setCount(count + 1)}>
+          {(count <= 3 && (
```

Comment:
Add only 1 button for count <=3, 4 and 5. Handle the text and onClick based on count

Reply:
done